theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -63,3 +108,319 @@ def repoclosures(repo, releases, version) {
+            fi
+        """
+
+        package_dir = sh(returnStdout: true, script: "find -maxdepth 1 -type d -not -name '.'").trim()
+    }
+
+    dir("${build_dir}/${package_dir}") {
+        if (pull_request) {
+            last_commit = sh(reutrnStdout: true, script: "git rev-list HEAD|/usr/bin/head -n 1").trim()
+            add_debian_changelog(os, package_version, repoowner, last_commit)
+        }
+    }
+
+    return "${build_dir}/${package_dir}"
+}
+
+def setup_sources_plugin(project, os, version, repoowner, pull_request = false) {
+    def package_version
+    def build_dir = deb_build_dir('plugin', project)
+    def package_dir
+
+    sh "mkdir -p ${build_dir}"
+
+    dir(build_dir) {
+        if (project.contains('smart_proxy')) {
+            package_version = sh(returnStdout: true, script: "head -n1 ../${project}/changelog|awk '{print \$2}'|sed 's/(//;s/)//'|cut -f1 -d-").trim()
+
+            sh """
+                gem fetch ${project} -v "=${package_version}"
+                ../../dependencies/gem2deb ${project}-${package_version}.gem --debian-subdir ../${project} --only-source-dir
+            """
+
+            package_dir = sh(returnStdout: true, script: "find -maxdepth 1 -type d -not -name '.'").trim()
+        } else {
+            package_version = sh(returnStdout: true, script: "head -n1 ../${project}/debian/changelog|awk '{print \$2}'|sed 's/(//;s/)//'|cut -f1 -d-|cut -d: -f2").trim()
+
+            sh """
+                BUILD_TYPE=gem
+                # Import variables from the project, allowing it to override behaviour
+                if [ -e ../${project}/build_vars.sh ]; then
+                  . ../${project}/build_vars.sh
+                fi
+
+                if [ "\$BUILD_TYPE" = "gem" ]; then
+                    cp -r ../${project} ./
+                    cd ${project}
+                    ../../download_gems
+                elif [ "\$BUILD_TYPE" = "ansible-collection" ]; then
+                    COLLECTION=${project.replace('ansible-collection-', '')}
+                    wget "https://galaxy.ansible.com/download/\${COLLECTION}-${package_version}.tar.gz"
+                    mv "\${COLLECTION}-${package_version}.tar.gz" "${project}_${package_version}.orig.tar.gz"
+                    cp -r ../${project} ./
+                    tar -x -C "${project}" -f "${project}_${package_version}.orig.tar.gz"
+                else
+                    echo "Unsupported build type: \${BUILD_TYPE}"
+                    exit 1
+                fi
+            """
+
+            package_dir = project
+        }
+    }
+
+    dir("${build_dir}/${package_dir}") {
+        if (pull_request) {
+            last_commit = sh(reutrnStdout: true, script: "git rev-list HEAD|/usr/bin/head -n 1").trim()
+            add_debian_changelog(os, package_version, repoowner, last_commit)
+        }
+    }
+
+    return "${build_dir}/${package_dir}"
+}
+
+def add_debian_changelog(os, package_version, repoowner, last_commit) {
```

Comment:
Attempting to make this change here first to reduce the diff https://github.com/theforeman/jenkins-jobs/pull/100

Reply:
This now contains the fix from #100